<?xml version="1.0" ?>
<launch>
  <arg name="gui" default="true" />
  <arg name="world_name" default="worlds/empty.world"/> <!-- Note: the world_name is with respect to GAZEBO_RESOURCE_PATH environmental variable, but can also be an absolute path -->
  <arg name="controller_name" default="none"/>
 <param name="frequenzy" value="10" />







<include file="$(find multi_robot_system)/launch/slave.launch">

  <arg name="robot_name" value="robot1"/>
  <arg name="robot_x"   value="1.25" />
  <arg name="robot_y"   value="1.25" />
  <arg name="name_master" value="robot_master"/>
  <arg name="displ"        value="screen"/>
</include>




<include file="$(find multi_robot_system)/launch/slave.launch">

  <arg name="robot_name" value="robot2"/>
  <arg name="robot_x"   value="1.25" />
  <arg name="robot_y"   value="-1.25" />
  <arg name="name_master" value="robot_master"/>
</include>




<include file="$(find multi_robot_system)/launch/slave.launch">
  <arg name="robot_name" value="robot3"/>
  <arg name="robot_x"   value="-0.75" />
  <arg name="robot_y"   value="-0.75" />
  <arg name="name_master" value="robot_master"/>
</include>


<include file="$(find multi_robot_system)/launch/slave.launch">
  <arg name="robot_name" value="robot4"/>
  <arg name="robot_x"   value="-0.75" />
  <arg name="robot_y"   value="0.75" />
  <arg name="name_master" value="robot_master"/>
</include>




<include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="world_name" value="$(arg world_name)"/>
      <arg name="paused" value="false" />
      <arg name="gui" value="$(arg gui)" />
</include>

  




<node name="PathPlaner" type="pathPlanner.py" pkg="multi_robot_system" args="/key_vel robot_master/master_input $(arg controller_name)"/>

<include file="$(find multi_robot_system)/launch/master.launch">
  <arg name="robot_name" value="robot_master"/>
  <arg name="robot_x"   value="0.0" />
  <arg name="robot_y"   value="0.0" />
  <arg name="slaves" value="robot1 robot2 robot3 robot4"/>
</include>

</launch>
